service/worker/deletenamespace/deleteexecutions/workflow.go

```diff
@@ -171,7 +171,9 @@ func DeleteExecutionsWorkflow(ctx workflow.Context, params DeleteExecutionsParam
 		}
 	}
 
-	if nextPageToken == nil {
+	// If nextPageToken is nil then there are no more workflow executions to delete.
+	// If SuccessCount == 0 then no progress were made with this run: exit here and let caller retry and eventually fail.
+	if nextPageToken == nil || result.SuccessCount == 0 {
```

Comment:
If success count is 0 and error count is not 0, we should continue?

Reply:
If both counters are 0, it means nothing was deleted. And if `nextPageToken` is not `nil` it means there is still something in visibility.

Reply:
I think my confusion is what happens if `nextPageToken != nil && SuccessCount == 0 && ErrorCount != 0`, like deletions fails for all workflows in the page case.

Reply:
if `nextPageToken` is not nil (there is something else left) it won't fall inside this if, and will do "continue as new". It will fall inside this if (and exit) only if there is nothing else left (`nextPageToken` is nil) or no progress were made (`SuccessCount` is 0). And this is what I explicitly said there in a comment.

Reply:
After second thought I think it is unneeded pre-caution. I removed `SuccessCount` check for now.